keanu-project/src/test/java/io/improbable/keanu/vertices/generic/nonprobabilistic/PrintVertexTest.java

```diff
@@ -0,0 +1,144 @@
+
+        assertThat(sut.getParents(), equalTo(ImmutableSet.of(parent)));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void whenCreatedWithNullParentThenFails() {
+        new PrintVertex<>(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void whenCreatedWithNullPrefixParentThenFails() {
+        final DoubleVertex parent = new ConstantDoubleVertex(30);
+        new PrintVertex<>(parent, null, false);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void whenPrintStreamIsNullThenFails() {
+        PrintVertex.setPrintStream(null);
+    }
+
+    @Test
+    public void whenOnlyParentIsSpecifiedThenDefaultPresentationIsUsed() {
+        final Vertex<DoubleTensor> parent = new ConstantDoubleVertex(30);
+
+        final PrintVertex<DoubleTensor> sut = new PrintVertex<>(parent);
+
+        sut.calculate();
+        final String expected = "Calculated Vertex:\n" +
+            "{\n" +
+            "data = [30.0]\n" +
+            "shape = []\n" +
+            "}\n";
+        verify(printStream).print(expected);
+    }
+
+    @Test
+    public void whenPrintVertexHasAChildThenChildCanGetGrandParentsValue() {
+        // this use case is quite unlikely, the Print vertex is meant to hang off its parent but not have any children
+        final Vertex<IntegerTensor> parent = new ConstantIntegerVertex(30);
+
+        final PrintVertex<IntegerTensor> sut = new PrintVertex<>(parent);
+
+
+        final UnaryOpVertex<IntegerTensor, IntegerTensor> child = new PlusOneOp(new long[]{1, 1}, sut);
+
+        assertThat(sut.getValue().scalar(), equalTo(30));
+        assertThat(child.getValue().scalar(), equalTo(31));
+    }
+
+    @Test
+    public void whenPrefixIsSuppliedThenItIsUsedInPresentation() {
+        final Vertex<DoubleTensor> parent = new ConstantDoubleVertex(30);
+
+        final PrintVertex<DoubleTensor> sut = new PrintVertex<>(parent, "my vertex\n", false);
+
+        sut.calculate();
+        final String expected = "my vertex\n";
+        verify(printStream).print(expected);
+    }
+
+    @Test
+    public void whenPrefixAndDataFlagAreSuppliedThenTheyAreUsedInPresentation() {
+        final Vertex<DoubleTensor> parent = new ConstantDoubleVertex(30);
+
+        final PrintVertex<DoubleTensor> sut = new PrintVertex<>(parent, "my vertex\n", true);
+
+        sut.calculate();
+        final String expected = "my vertex\n" +
+            "{\n" +
+            "data = [30.0]\n" +
+            "shape = []\n" +
+            "}\n";
+        verify(printStream).print(expected);
+    }
+
+    @Test
+    public void whenRunningMetropolisHastingsThenSamplesArePrinted() {
+        final Vertex<DoubleTensor> temperature = new UniformVertex(20., 30.);
+
+        new PrintVertex<>(temperature);
+
+        final BayesianNetwork bayesNet = new BayesianNetwork(temperature.getConnectedGraph());
+
+        final int nSamples = 10000;
```

Comment:
Can you lower this sample amount? We can verify the functionality with 10 samples and it will mean the tests run quicker.